agent/agent.go

```diff
@@ -199,15 +219,36 @@ func (a *Agent) startCheckTicker() error {
 	}
 }
 
+func (a *Agent) startDiscoverTicker() error {
+	ticker := time.NewTicker(a.cfg.DiscoverInterval / 2)
+	// Repeat after 15 minutes
+	if a.cfg.DiscoverInterval < time.Minute {
+		a.cfg.DiscoverInterval = 15 * time.Minute
+	}
+	defer ticker.Stop()
+	for {
+		select {
+		case <-ticker.C:
+			for i := 0; i < len(a.discovery); i++ {
```

Comment:
I don't really understand this.
What is the purpose of having a list of discoveries? How is this going to evolve in the future?

Reply:
The idea is that we only run "discoveries" where it makes sense, e.g. we only collect Hana specific data on nodes that have either been tagged as such or are matching the detection that this is a hana node. 

at some point we can have a configurable list of "introspections" (Discoveries) to be executed by node, which is then filtered by "is this filter configured *and* does it match the expectation", and then only the configured collectors that apply are executed in the discover loop. 

For now there is only one.